### PR TITLE
Added small fix to scientific notation string formatting

### PR DESCRIPTION
**Pull request by darkaardvark2**
Previously, the following statement 
`print("{:e}".format(123456789))`
would show
`1.234568e+8`

Now fixed power number padding to show
`1.234568e+08`

### CODE REVIEW

File: src/formatting.js [hunk 1]
```diff
@@ -189,6 +189,16 @@ let formatNumber = function(num, formatSpec, isFractional) {
                     result += ".0";
                 }
             }
+            // Python's 'e' pads the power with 0 if it is a single digit
+            if (conversionType.toLowerCase()==="e") {
```
Comment (meredydd) [inline src/formatting.js]: We should probably also handle the `g` conversion type too!

File: src/formatting.js [hunk 2]
```diff
@@ -189,6 +189,16 @@ let formatNumber = function(num, formatSpec, isFractional) {
...
+                let scientificNotation = result.match(/[-+]?[0-9]*\.?[0-9]+[eE][-+]?([0-9]+)?/);
```
Comment (meredydd) [inline src/formatting.js]: Code-size nit: Can we collapse this into one `replace()` call? Something like:

```javascript
result = result.replace(/^([-+]?[0-9]*\.?[0-9]+[eE][-+]?)([0-9])?$/, "$10$2");
```

**Comment by rixner** [comment]
Can we get this updated so that we can get it merged?

**Comment by meredydd** [comment]
Hi @darkaardvark2 - would you be able to update this PR? We'd love to get this merged!

**Comment by s-cork** [comment]
closed via #1226

**Pull request closed without merging**